src/app/machines/components/TableCheckbox/TableCheckbox.tsx

```diff
@@ -49,10 +52,17 @@ const TableCheckbox = ({
       disabled={machineCount === 0 || isDisabled}
       label={inputLabel}
       labelClassName="u-no-margin--bottom u-no-padding--top"
-      onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+      onChange={(
+        event: React.ChangeEvent<HTMLInputElement> & {
+          nativeEvent: React.PointerEvent;
+        }
+      ) => {
+        // prevent default text range selection when 'shift' key is pressed
+        window.getSelection()?.removeAllRanges();
+        const isRange = !!event.nativeEvent.shiftKey;
```

Comment:
As described in the comment, we should narrow down overriding the default behaviour to only when the shift key is pressed.

```suggestion
        // prevent default text range selection when 'shift' key is pressed
        const isRange = !!event.nativeEvent.shiftKey;
        if (isRange) {
          window.getSelection()?.removeAllRanges();
        }
```

Reply:
I had tried this earlier, but the check doesn't occur fast enough to prevent the text selection behaviour, this was why I took it to the top level

Reply:
I'll give it another go and let you know if it works

Reply:
In that case you might need to place this in onClickCapture instead to check this during the event capture phase.

Reply:
If it's causing you too much trouble I'm happy with leaving it as is, in theory we wouldn't ever want to keep the selection when you click on a checkbox.